cfme/infrastructure/cluster.py

```diff
@@ -318,6 +319,28 @@ def delete(self, *clusters):
         for cluster in clusters:
             cluster.wait_for_disappear()
 
+    def all(self):
+        """returning all cluster objects and support filtering as per provider"""
+        cluster_obj = []
+        provider = self.filters.get("provider")
+        clusters = self.appliance.rest_api.collections.clusters.all
+        if provider:
+            cluster_obj = [
+                self.instantiate(name=cluster.name, provider=provider)
+                for cluster in clusters
+                if provider.id == cluster.ems_id
+            ]
+        else:
+            providers = self.appliance.rest_api.collections.providers
```

Comment:
It could be beneficial to create a database of providers and their ids, so you don't need to repeat the ``.find_by`` for each ``cluster``. Like
```python
providers_db = {prov.id: get_crud_by_name(prov.name) for prov in providers}
cluster_obj = [self.instantiate(name=cluster.name, provider=providers_db[cluster.ems_id]) for cluster in clusters]
```